api_app/api/routes/costs.py

```diff
@@ -13,41 +15,36 @@
 from models.domain.costs import CostReport, GranularityEnum, WorkspaceCostReport
 from resources import strings
 from services.authentication import get_current_admin_user, get_current_workspace_owner_or_tre_admin
-from services.cost_service import CostService
+from services.cost_service import CostService, WorkspaceDoesNotExist
 
 costs_core_router = APIRouter(dependencies=[Depends(get_current_admin_user)])
 costs_workspace_router = APIRouter(dependencies=[Depends(get_current_workspace_owner_or_tre_admin)])
 
 
-def get_first_day_of_month():
-    return date_to_datetime(date.today().replace(day=1))
-
-
-def get_tomorrow_datetime():
-    return date_to_datetime(datetime.today() + relativedelta(days=1))
-
-
-def date_to_datetime(date_to_covert: date):
-    converted_datetime = datetime.combine(date_to_covert, datetime.min.time())
-    converted_datetime.replace(tzinfo=pytz.UTC)
-    return converted_datetime
+def validate_report_period(from_date: Optional[datetime], to_date: Optional[datetime]):
+    if from_date is None and to_date is None:
+        # valid option, month to date report
+        return
 
-
-def validate_report_period(from_date: datetime, to_date: datetime):
+    if to_date is None:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST,
+                            detail=strings.API_GET_COSTS_TO_DATE_NEED_TO_BE_LATER_THEN_FROM_DATE)
+    if from_date is None:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST,
+                            detail=strings.API_GET_COSTS_FROM_DATE_NEED_TO_BE_BEFORE_TO_DATE)
     if relativedelta(to_date, from_date).years > 0:
         raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=strings.API_GET_COSTS_MAX_TIME_PERIOD)
-    elif from_date >= to_date:
+
+    if from_date >= to_date:
         raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST,
                             detail=strings.API_GET_COSTS_TO_DATE_NEED_TO_BE_LATER_THEN_FROM_DATE)
 
 
 class CostsQueryParams:
     def __init__(
         self,
-        from_date: datetime = Query(default=get_first_day_of_month(),
-                                    description="The start date to pull data from, default value first day of month (iso-8601, UTC)."),
-        to_date: datetime = Query(default=get_tomorrow_datetime(),
-                                  description="The end date to pull data to, default tomorrow's date (iso-8601, UTC)."),
+        from_date: Optional[datetime] = Query(default=None, description="The start date to pull data from, requiered if to_date is set, otherwise report will return month to date (iso-8601, UTC)."),
```

Comment:
default=None is redundant?

Reply:
TypeError: Query() missing 1 required positional argument: 'default'